tasks/Task.cpp

```diff
@@ -0,0 +1,152 @@
+    std::string errs;
+
+    if (!reader->parse(data, data+std::strlen(data), &jdata, &errs)){
+        std::cout << errs << std::endl;
+        return false;
+    }
+    if (!(jdata.isMember("status") && jdata.isMember("axes") && jdata.isMember("buttons"))) {
+        return false;
+    }
+    try{
+        controlling = jdata["status"].asBool();
+
+        raw_cmd_obj.axisValue[0] = jdata["axes"][0].asDouble();
+        raw_cmd_obj.axisValue[1] = jdata["axes"][1].asDouble();
+        raw_cmd_obj.axisValue[2] = jdata["buttons"][6].asDouble() * 2 - 1;
+        raw_cmd_obj.axisValue[3] = jdata["axes"][2].asDouble();
+        raw_cmd_obj.axisValue[4] = jdata["axes"][3].asDouble();
+        raw_cmd_obj.axisValue[5] = jdata["buttons"][7].asDouble() * 2 - 1;
+        raw_cmd_obj.axisValue[6] = jdata["buttons"][15].asDouble() - jdata["buttons"][14].asDouble();
+        raw_cmd_obj.axisValue[7] = jdata["buttons"][13].asDouble() - jdata["buttons"][12].asDouble();
+
+        raw_cmd_obj.buttonValue[0] = jdata["buttons"][0].asDouble() > _button_threshold.get();
+        raw_cmd_obj.buttonValue[1] = jdata["buttons"][1].asDouble() > _button_threshold.get();
+        raw_cmd_obj.buttonValue[2] = jdata["buttons"][2].asDouble() > _button_threshold.get();
+        raw_cmd_obj.buttonValue[3] = jdata["buttons"][3].asDouble() > _button_threshold.get();
+        raw_cmd_obj.buttonValue[4] = jdata["buttons"][4].asDouble() > _button_threshold.get();
+        raw_cmd_obj.buttonValue[5] = jdata["buttons"][5].asDouble() > _button_threshold.get();
+        raw_cmd_obj.buttonValue[6] = jdata["buttons"][8].asDouble() > _button_threshold.get();
+        raw_cmd_obj.buttonValue[7] = jdata["buttons"][9].asDouble() > _button_threshold.get();
+
+    } catch (std::exception e){
+        std::cout << e.what() << std::endl;
+        return false;
+    }
+    return true;
+}
+
+Task::Task(std::string const& name, TaskCore::TaskState initial_state)
+    : TaskBase(name, initial_state)
+{
+    for (int i=1; i<=8; ++i){
+        raw_cmd_obj.buttonValue.push_back(0);
+        raw_cmd_obj.axisValue.push_back(0.0);
+    }
+}
+
+Task::~Task()
+{
+}
+
+bool Task::configureHook()
+{
+    if (! TaskBase::configureHook()){
+        return false;
+    }
+
+    auto logger = std::make_shared<PrintfLogger>(Logger::Level::Debug);
+    server = new Server (logger);
+    handler = std::make_shared<JoystickHandler>();
+    handler->task = this;
+
+    return true;
+}
+bool Task::startHook()
+{
+    if (! TaskBase::startHook()){
+        return false;
+    }
+
+    server->addWebSocketHandler("/ws", handler, true);
+    if (!server->startListening(_port.get())) {
+        return false;
+    }
+
+    thread = new std::thread([this]{ this->server->loop(); });
+
+    return true;
+    return true;
+}
+void Task::updateHook()
+{
+    TaskBase::updateHook();
+}
+void Task::errorHook()
+{
+    TaskBase::errorHook();
+    server->terminate();
```

Comment:
`errorHook` is a **runtime** error, i.e. it is meant to go back to normal operations afterwards.

The hook 'on error' is `exceptionHook`, which automatically goes through stopHook and cleanupHook anyways. So just do cleanup in those.